algorithms/src/polycommit/kzg10/data_structures.rs

```diff
@@ -108,6 +134,118 @@ impl<E: PairingEngine> UniversalParams<E> {
             prepared_negative_powers_of_beta_h: self.powers.read().prepared_negative_powers_of_beta_h(),
         })
     }
+
+    pub fn to_committer_key(
```

Comment:
Is this a port of the `CommitterUnionKey` code?

Reply:
Seems like a combination of that + `trim`

Reply:
It is a port of `trim`, the union structs and logic are complete superfluous in this new setup.